src/dir.targets

```diff
@@ -51,10 +56,25 @@
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets"
           Condition="'$(TargetFrameworkIdentifier)' != '.NETPortable'" />
 
+  <!-- Restore packages -->
+  <PropertyGroup>
+    <PackagesConfig Condition="Exists('$(MSBuildProjectDirectory)\packages.config')">$(MSBuildProjectDirectory)\packages.config</PackagesConfig>
+    <RestorePackages Condition="'$(RestorePackages)' == '' and Exists('$(PackagesConfig)')">true</RestorePackages>
+    <RestorePackagesSemaphore>$(PackagesDir)$(MSBuildProjectFile).package.restored</RestorePackagesSemaphore>
+  </PropertyGroup>
+  <Target Name="RestorePackages" 
+          BeforeTargets="ResolveNuGetPackages"
+          Inputs="$(PackagesConfig)"
+          Outputs="$(RestorePackagesSemaphore)"
+          Condition="'$(RestorePackages)' == 'true'">
+    <Exec Command="$(NugetRestoreCommand) &quot;$(PackagesConfig)&quot;" StandardOutputImportance="Low" />
+    <Touch Files="$(RestorePackagesSemaphore)" AlwaysCreate="true" />
```

Comment:
Why stop package restore from running on incremental build? If all of the packages are already present then I would expect package restore to take a negligible amount of time. If any package was deleted, but the semaphore file was not, the build will fail.